scripts/step03_explore_examples/06_rag_benchmarking/src/evaluate_without_golden_testset.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from rag_benchmark_utils.common_utils import create_retriever, validate_response, print_results
+from helpers.config import TERRAFORM_DOCS_TABLE_NAME
+
+log = logging.getLogger(__name__)
+
+def evaluate_without_golden_testset():
+    """
+    Evaluates the RAG application without using a golden test set.
+    """
+    # Define test queries
+    test_cases = [
+        {"query": "What is Terraform?"},
+        {"query": "How to use Terraform with SAP?"},
+        {"query": "Explain the concept of Infrastructure as Code."},
+        {"query": "What are the benefits of using Terraform?"},
+        {"query": "How to set up a Terraform provider for SAP BTP?"},
+    ]
+
+    qa_chain = create_retriever(TERRAFORM_DOCS_TABLE_NAME)
+    results = []
+
+    for case in test_cases:
+        query = case["query"]
+
+        try:
+            log.info(f"Asking question: {query}")
+            result = qa_chain.invoke({"query": query})
+            actual_answer = result.get("result", "N/A")
```

Comment:
Again not a must but constants for N/A and Failed would be better in this scenario